lib/miner-info.js

```diff
@@ -29,6 +30,32 @@ async function getChainHead ({ maxAttempts = 5 } = {}) {
  * @returns {Promise<string>} Miner's PeerId, e.g. `12D3KooWMsPmAA65yHAHgbxgh7CPkEctJHZMeM3rAvoW8CZKxtpG`
  */
 export async function getMinerPeerId (minerId, { maxAttempts = 5 } = {}) {
+  // Make a concurrent request to both sources: RPC node and smart contract
+  const [rpcNodeResult, contractResult] = await Promise.allSettled([
+    getMinerPeerIdFromRpcNode(minerId, { maxAttempts }),
+    getMinerPeerIdFromSmartContract(minerId)
+  ])
+  // Check contract result first
+  if (contractResult.status === 'fulfilled' && contractResult.value) {
+    return contractResult.value
```

Comment:
Let's add a `console.log` for diagnostics.

```suggestion
    console.log('Using PeerID from the smart contract.')
    return contractResult.value
```

Reply:
66a2d4a